File: app/src/main/java/io/github/pierry/better_call_me/ui/adapters/InfoAdapter.java
package io.github.pierry.better_call_me.ui.adapters;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import io.github.pierry.better_call_me.R;
import io.github.pierry.better_call_me.domain.viewmodels.Data;
import io.github.pierry.better_call_me.ui.holders.InfoHolder;
import java.util.List;

public class InfoAdapter extends RecyclerView.Adapter<InfoHolder> {

  private List<Data> items;

  public void addItems(List<Data> items) {
    this.items = items;
  }

  @NonNull @Override public InfoHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
    View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.data_adapter, parent, false);
    return new InfoHolder(itemView);
  }

  @Override public void onBindViewHolder(@NonNull InfoHolder holder, int position) {
    Data data = items.get(position);
    holder.bind(data);
  }

  @Override public int getItemCount() {
    return items != null ? items.size() : 0;
  }
}

File: SantanderInvestimento/app/src/main/kotlin/br/com/santander/santanderinvestimento/feature/investiment/adapter/InfoAdapter.kt
package br.com.santander.santanderinvestimento.feature.investiment.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import br.com.santander.santanderinvestimento.R
import br.com.santander.santanderinvestimento.feature.investiment.domain.entity.Info
import kotlinx.android.synthetic.main.item_view_info.view.*

class InfoAdapter(listItems: MutableList<Info>) : RecyclerView.Adapter<InfoAdapter.InfoViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): InfoViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_view_info, parent, false)
        return InfoViewHolder(view)
    }

    private val infoListItems: MutableList<Info> = listItems


    override fun getItemCount() = infoListItems.size

    override fun onBindViewHolder(holder: InfoViewHolder, position: Int) {
        val note = infoListItems[position]
        holder.bindView(note)
    }

    inner class InfoViewHolder(view: View) : RecyclerView.ViewHolder(view) {

        fun bindView(item: Info) {
            setIsRecyclable(false)
            itemView.txName.text = item.name
            itemView.txTax.text = item.data
        }
    }
}